### DISCUSSION TITLE
Customize index.html

### DISCUSSION THREAD

**Discussion by root9191** [authelia/authelia#4407] [Q&A]
Hello,
is it possible to customize the index.html? I wanna integrate matomo in the Header, so i can see how much User are on this Site and where they are from. I could handle to change the Logo and the Favicon but i don't know how i can implement the Code of matomo here.

**Comment by james-d-elliott** [comment]
There is no way to do this at the present time. We can consider adding a method to inject specific tags into index.html. Can you show an example of what you'd like to change?

**Comment by root9191** [comment]
Hi,
So this is the Code that needs to get implemented in the Header: 
```
<!-- Matomo -->
<script>
  var _paq = window._paq = window._paq || [];
  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="https://DOMAIN.com/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
<!-- End Matomo Code -->
```

This is the Website of Matomo: [https://matomo.org/](url)
It is like Google Analytics and tracks the Visitors on my Page.

**Reply by nightah** [reply]
If you utilise NGINX as your reverse proxy, you can do this with the [`sub_filter` directive](https://nginx.org/en/docs/http/ngx_http_sub_module.html#sub_filter).

**Reply by root9191** [reply]
Could you explain how i can set this right here?

**Reply by root9191** [reply]
Authelia runs behind the NGINX Proxy Manager, but Matomo is hosted on a Webhosting Platform.

**Comment by markus-li** [comment]
Just like @nightah mentions using sub_filter works well, but it'd be great if it was possible to configure a JS and CSS URL which would be added to the html. In my opinion Authelia should not need to care about where those files are hosted from, just create the tags to include them.
One of the two other parts I use sub_filter for is rewriting base href to not include the port in the URL. I'm serving Authelia on two different ports, so this is not a normal requirement I'm sure.
The last part I use sub_filter for is setting a custom title of the page, eg. instead of "Login - Authelia" it shows "Login - SSO".
Just wanted to add some input as to the reasoning and level of customization may be used by others, personally I'm happy to continue using sub_filter.
